### PR TITLE
HDDS-2849. OM and SCM Web-server report HTTP 404 error when accessing…

### PR DESCRIPTION
**Pull request by ChenSammi**
https://issues.apache.org/jira/browse/HDDS-2849

### CODE REVIEW

File: hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/server/BaseHttpServer.java [hunk 1]
```diff
@@ -114,8 +118,13 @@ public BaseHttpServer(Configuration conf, String name) throws IOException {
                 + "production!");
         httpServer.addServlet("profile", "/prof", ProfileServlet.class);
       }
-    }
 
+      String baseDir = conf.get(OzoneConfigKeys.OZONE_HTTP_BASEDIR);
+      if (!StringUtils.isEmpty(baseDir)) {
+        httpServer.getWebAppContext().setAttribute(JETTY_BASETMPDIR, baseDir);
```
Comment (xiaoyuyao) [inline hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/server/BaseHttpServer.java]: Thanks @ChenSammi  for reporting the issue and working on the fix. 

According to jetty document [here](https://www.eclipse.org/jetty/documentation/current/ref-temporary-directories.html): 

"By default, Jetty will create this directory inside the directory named by the java.io.tmpdir System property. You can instruct Jetty to use a different parent directory by setting the context attribute org.eclipse.jetty.webapp.basetempdir to the name of the desired parent directory. The directory named by this attribute must exist and be writeable."

Given the behavior without specifying basetmpdir explicitly, jetty will automatically create the dir under java.io.tempdir, which should work. Can you elaborate on the proposed fix?

Comment (elek) [reply hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/server/BaseHttpServer.java]: From an offline discussion I learned the problem: In some environments the `/tmp` cleaned up regularly (For me it's just at the restart but in some environments it's cleaned up more frequently). In these environments the web UI is not available after a while (when jetty temporary files are deleted).
 
I am fine with this patch. We can remove this configuration setting long-term as we are planning to unify all the webui under the recon. Until that we can make it configurable...

Comment (elek) [reply hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/server/BaseHttpServer.java]: Let me know if you are happy with this approach @xiaoyuyao and I will commit the patch...

Comment (xiaoyuyao) [reply hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/server/BaseHttpServer.java]: Sounds good to me. Agree that could be a problem when /tmp is cleaned up more frequently.

**Pull request closed without merging**